### PR TITLE
Initial version of the /hwinfo API

### PR DESCRIPTION
**Pull request by dangogh**

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/348/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR-trafficops-test/306/

File: traffic_ops/tostructs/hwinfos.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+type HwInfoResponse struct {
```
Comment (dewrich) [inline traffic_ops/tostructs/hwinfos.go]: "Idiomatic Go" says all acronyms are capitalized, for example: HWInfoResponse

File: traffic_ops/tostructs/hwinfos.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	Response []HwInfo `json:"response"`
+}
+
+type HwInfo struct {
```
Comment (dewrich) [inline traffic_ops/tostructs/hwinfos.go]: We've been keeping the struct fields alphabetized up to these point, not sure what the feeling is for that as a whole

**Comment by dangogh** [comment]
addresses part of #1306

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/371/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/372/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/373/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/374/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR-trafficops-test/329/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR-trafficops-test/330/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR-trafficops-test/331/

**Pull request merged**